addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/handler/PlugwiseRelayDeviceHandler.java

```diff
@@ -0,0 +1,576 @@
+        @Override
+        public int getConfiguredInterval() {
+            return getChannelUpdateInterval(CHANNEL_ENERGY);
+        }
+
+        @Override
+        public void runTask() {
+            if (isRecentLogAddressKnown()) {
+                updateEnergy();
+            }
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && isLinked(CHANNEL_ENERGY);
+        }
+    };
+
+    private final PlugwiseDeviceTask informationUpdateTask = new PlugwiseDeviceTask("Information update", scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return Math.min(getChannelUpdateInterval(CHANNEL_STATE), getChannelUpdateInterval(CHANNEL_ENERGY));
+        }
+
+        @Override
+        public void runTask() {
+            updateInformation();
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && (isLinked(CHANNEL_STATE) || isLinked(CHANNEL_ENERGY));
+        }
+    };
+
+    private final PlugwiseDeviceTask realTimeClockUpdateTask = new PlugwiseDeviceTask("Real-time clock update",
+            scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return getChannelUpdateInterval(CHANNEL_REAL_TIME_CLOCK);
+        }
+
+        @Override
+        public void runTask() {
+            sendMessage(new RealTimeClockGetRequestMessage(macAddress));
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && deviceType == DeviceType.CIRCLE_PLUS
+                    && isLinked(CHANNEL_REAL_TIME_CLOCK);
+        }
+    };
+
+    private final PlugwiseDeviceTask setClockTask = new PlugwiseDeviceTask("Set clock", scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return 24 * 60 * 60; // Once every 24 hours;
+        }
+
+        @Override
+        public void runTask() {
+            sendMessage(new ClockSetRequestMessage(macAddress, LocalDateTime.now()));
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && deviceType == DeviceType.CIRCLE_PLUS;
+        }
+    };
+
+    private final List<PlugwiseDeviceTask> recurringTasks = Lists.newArrayList(clockUpdateTask, currentPowerUpdateTask,
+            energyUpdateTask, informationUpdateTask, realTimeClockUpdateTask, setClockTask);
+
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseRelayDeviceHandler.class);
+
+    private PlugwiseRelayConfig configuration;
+    private DeviceType deviceType;
+    private MACAddress macAddress;
+
+    private PowerCalibration calibration;
+    private Energy energy;
+    private int recentLogAddress = -1;
+    private PendingPowerStateChange pendingPowerStateChange;
+
+    // Configuration update command to send when the device is online
```

Comment:
The variable below the comment is a flag that keeps track of whether or not a command should be send to update the measurement interval of the device. So the flag remains true unless the device sends an ACK that acknowledges that the measurement interval in the device was successfully changed.